gax-grpc/src/test/java/com/google/api/gax/grpc/GrpcDirectServerStreamingCallableTest.java

```diff
@@ -106,24 +115,116 @@ public void testBadContext() {
+
+    streamingCallable.call(DEFAULT_REQUEST, moneyObserver);
+
+    latch.await(500, TimeUnit.MILLISECONDS);
+    Truth.assertWithMessage("Received response before requesting it")
+        .that(moneyObserver.response)
+        .isNull();
+
+    moneyObserver.controller.request(1);
+    latch.await(500, TimeUnit.MILLISECONDS);
+
+    Truth.assertThat(moneyObserver.response).isEqualTo(DEFAULT_RESPONSE);
+    Truth.assertThat(moneyObserver.completed).isTrue();
+  }
+
+  @Test
+  public void testCancelClientCall() throws Exception {
+    CountDownLatch latch = new CountDownLatch(1);
+    MoneyObserver moneyObserver = new MoneyObserver(false, latch);
+    CancellationException expectedCause = new CancellationException("cancelled!");
+
+    streamingCallable.call(ASYNC_REQUEST, moneyObserver);
+
+    moneyObserver.controller.cancel(expectedCause);
+    moneyObserver.controller.request(1);
+    latch.await(500, TimeUnit.MILLISECONDS);
+
+    Truth.assertThat(moneyObserver.error).isInstanceOf(StatusRuntimeException.class);
+    Truth.assertThat(((StatusRuntimeException) moneyObserver.error).getStatus().getCode())
+        .isEqualTo(Status.CANCELLED.getCode());
+    Truth.assertThat(moneyObserver.error.getCause()).isSameAs(expectedCause);
+  }
+
+  @Test
+  public void testOnResponseError() throws Throwable {
+    CountDownLatch latch = new CountDownLatch(1);
+    MoneyObserver moneyObserver = new MoneyObserver(true, latch);
+
+    streamingCallable.call(ERROR_REQUEST, moneyObserver);
+    latch.await(500, TimeUnit.MILLISECONDS);
+
+    Truth.assertThat(moneyObserver.error).isInstanceOf(StatusRuntimeException.class);
+    Truth.assertThat(moneyObserver.error).hasMessage("INVALID_ARGUMENT: red must be positive");
+  }
+
+  @Test
+  public void testObserverErrorCancelsCall() throws Throwable {
+    final RuntimeException expectedCause = new RuntimeException("some error");
+    final SettableApiFuture<Throwable> actualErrorF = SettableApiFuture.create();
+
+    ResponseObserver<Money> moneyObserver =
+        new ResponseObserver<Money>() {
+          @Override
+          public void onStart(StreamController controller) {}
+
+          @Override
+          public void onResponse(Money response) {
+            throw expectedCause;
+          }
+
+          @Override
+          public void onError(Throwable t) {
+            actualErrorF.set(t);
+          }
+
+          @Override
+          public void onComplete() {
+            actualErrorF.set(null);
+          }
+        };
+
+    streamingCallable.call(DEFAULT_REQUEST, moneyObserver);
+    Throwable actualError = actualErrorF.get(500, TimeUnit.MILLISECONDS);
+
+    Truth.assertThat(actualError).isInstanceOf(StatusRuntimeException.class);
+    Truth.assertThat(((StatusRuntimeException) actualError).getStatus().getCode())
+        .isEqualTo(Status.CANCELLED.getCode());
+    Truth.assertThat(actualError.getCause()).isSameAs(expectedCause);
```

Comment:
This has similar behavior to cancel() in that the error that gets passed to onError is a StatusRuntimeException/ApiException that wraps the actual error. I'm wondering if the actual error should be the topmost.

Reply:
I think keeping current behavior is OK

Reply:
SGTM